### PR TITLE
[installer]: put component ownership under webapp/workspace teams

### LINKED ISSUE #7690: [installer] Make the Kubernetes objects subject to per-team approval
When making changes to components in the installer today, one needs the approval of the team which owns that component. That does not hold for the set of objects that make up the environment, e.g.

https://github.com/gitpod-io/gitpod/blob/e6924d94369df23af8611a33f59a93ebdb67e76b/installer/pkg/components/components.go#L35-L64

This has broken things in the past, e.g. https://github.com/gitpod-io/gitpod/issues/7686

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Create a `components-webapp` and `components-workspace` package to make the Kubernetes objects subject to per-team approval.

A couple of observations as to why I did it the way I did:
 - I originally wanted to put it in format `/pkg/components/workspace/<packages>`, but this caused cyclic dependency issues. The use of a `components-<package>` is a way of avoiding this, even though it's arguably a less elegant solution. I'm open to ideas on alternative ways of achieving this
 - All the `components-workspace` packages are owned by @gitpod-io/engineering-workspace. The `components-webapp` packages are owned by different teams, but I have elected to make the folder owned solely by @gitpod-io/engineering-webapp
 - The common components remain owned by @gitpod-io/engineering-self-hosted, so no changes are made here

## Related Issue(s)

Fixes #7690

## Release Notes

```release-note
[installer]: put component ownership under webapp/workspace teams
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#7690*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by MrSimonEmms** [comment]
/hold

Approval is required from @gitpod-io/engineering-self-hosted to go in but would like @csweichel to give his overview before it's merged in

File: installer/pkg/components/components-workspace/OWNERS
Comment (princerachit) [inline installer/pkg/components/components-workspace/OWNERS:9]: nit: new line. And other files

**Review by princerachit — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 7a4d510a8130774cbc9e65c79c7c2ab040b36328</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7839#pullrequestreview-868918587" title="Approved">princerachit</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#7690*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by princerachit** [comment]
looks good from workspace perspective

File: installer/pkg/components/components-workspace/OWNERS
Comment (MrSimonEmms) [reply installer/pkg/components/components-workspace/OWNERS:9]: Ordinarily I'd agree, but this is the case throughout the `OWNERS` files - in fact, this is copied directly over from them

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7839#issuecomment-1026690220" title="Approved">MrSimonEmms</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7839#pullrequestreview-868918587" title="Approved">princerachit</a>*

Associated issue: *#7690*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [MrSimonEmms]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
